hack/for-go-proj.sh

```diff
@@ -43,9 +43,6 @@ esac
 
 for project_name in ${PROJECT_NAMES[*]}; do
   (
-    if [[ $project_name == cluster-autoscaler ]];then
```

Comment:
The previous value used was "auto" for every project but cluster-autoscaler, and "off" for cluster-autoscaler, right? And now we're setting it to "off" for every project but vertical-pod-autoscaler/e2e, for which we're setting it to "on". So the delta is:
cluster-autoscaler: off -> off
vertical-pod-autoscaler/e2e: auto -> on
everything else: auto -> off

If we want to just make it work like before, shouldn't we be aiming for no delta? Also, if that's the goal (as opposed to updating the scripts), won't setting GO111MODULE=auto for all of the scripts be the easiest option? From the go docs you linked:
>More precisely, the GO111MODULE environment variable now defaults to on. To switch to the previous behavior, set GO111MODULE to auto.

Reply:
Great catch.
Changes this replicate earlier behaviour of using `auto`